api/clients/grailfiltersegements/filtersegments.go

```diff
@@ -0,0 +1,112 @@
+// @license
+// Copyright 2024 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grailfiltersegements
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+)
+
+const endpointPath = "platform/storage/filter-segments/v1/filter-segments"
+
+type Client struct {
+	client *rest.Client
+}
+
+func NewClient(client *rest.Client) *Client {
+	c := &Client{
+		client: client,
+	}
+	return c
+}
+
+// List gets the list of existing filter segments. To get as minimal necessary set of data it uses ":lean" endpoint. If query parameter "add-fields" is not specified, it will be set with next values: "EXTERNALID".
+func (c Client) List(ctx context.Context, ro rest.RequestOptions) (*http.Response, error) {
+	path := endpointPath + ":lean" // minimal set of information is enough
+
+	// set default behavior to request for ExternalID
+	if !ro.QueryParams.Has("add-fields") {
+		ro.QueryParams = url.Values{"add-fields": []string{"EXTERNALID"}}
+	}
+
+	r, err := c.client.GET(ctx, path, ro)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the list of filter segments: %w", err)
+	}
+
+	return r, nil
+}
+
+// Get filter-segment by UID. If query parameter "add-fields" is not specified, it will be set with next values: "INCLUDES", "VARIABLES", "EXTERNALID", "RESOURCECONTEXT".
```

Comment:
```suggestion
// // Get filter-segment by its ID.
// If the field 'add-fields' in [rest.RequestOptions.QueryParams] is not specified, it will be set to "INCLUDES", "VARIABLES", "EXTERNALID", "RESOURCECONTEXT".
```